application/application.go

```diff
@@ -0,0 +1,101 @@
+package application
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/josephroberts/edge-node-manager/config"
+	"github.com/josephroberts/edge-node-manager/database"
+	"github.com/josephroberts/edge-node-manager/device"
+	"github.com/josephroberts/edge-node-manager/helper"
+	"github.com/josephroberts/edge-node-manager/proxyvisor"
+)
+
+type Application struct {
+	UUID       string
+	DeviceType *device.Type
+}
+
+func (a *Application) Process() error {
+	log.Printf("-----------------------------------------------------------------\r\n")
+	log.Printf("Application UUID: %s\r\n", a.UUID)
+	log.Printf("Application device type: %s\r\n", a.DeviceType.Device)
+	log.Printf("Application radio type: %s\r\n", a.DeviceType.Radio)
+
+	// Get latest firmware .zip and commit
+	application, commit, err := helper.GetApplication(config.GetPersistantDirectory(), a.UUID)
```

Comment:
Since you're seemingly throwing away these values immediately (I'm not sure why you're doing this?) you may as well use a `if application, commit, err := helper.GetApplication(config.GetPersistantDirectory(), a.UUID); err != nil { ... } else { ... }` form. Having `application` and `commit` in scope here along with an `a` value that's meant to represent the application is confusing and in general I'd recommend using a different name.

I'm a little confused about how a `Process()` method on an application is immediately calling `GetApplication()`? What application? Surely it's already an appplication since the type is already instantiated?